homeassistant/components/srp_energy/sensor.py

```diff
@@ -68,78 +88,67 @@ async def async_update_data():
 
     coordinator = DataUpdateCoordinator(
         hass,
-        _LOGGER,
+        LOGGER,
         name="sensor",
         update_method=async_update_data,
         update_interval=MIN_TIME_BETWEEN_UPDATES,
     )
 
     # Fetch initial data so we have data when entities subscribe
-    await coordinator.async_refresh()
+    await coordinator.async_config_entry_first_refresh()
 
-    async_add_entities([SrpEntity(coordinator)])
+    async_add_entities([SrpEntity(coordinator, entry_name)])
 
 
 class SrpEntity(SensorEntity):
     """Implementation of a Srp Energy Usage sensor."""
 
+    _attr_has_entity_name = True
     _attr_attribution = ATTRIBUTION
     _attr_should_poll = False
 
-    def __init__(self, coordinator):
+    def __init__(self, coordinator, entry_name) -> None:
         """Initialize the SrpEntity class."""
         self._name = SENSOR_NAME
         self.type = SENSOR_TYPE
         self.coordinator = coordinator
         self._unit_of_measurement = UnitOfEnergy.KILO_WATT_HOUR
         self._state = None
+        self._entry_name = entry_name
+        unique_id = f"{entry_name}_energy_usage".lower()
```

Comment:
The unique_id must be unique and not include a name set by the user or non stable name.

Also, we should not add new features in this PR.

https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements

```suggestion
```